src/programs.c

```diff
@@ -181,11 +272,61 @@ static __always_inline syscall_pattern_t ptrace_syscall_pattern(u32 request)
         .mono_ns = mono_ns,                    \
     }
 
+#define FILL_TELEMETRY_SYSCALL_EVENT(E, SP)    \
+    u64 pid_tgid = bpf_get_current_pid_tgid(); \
+    E->done = FALSE;                           \
+    E->telemetry_type = TE_SYSCALL_INFO;       \
+    E->u.syscall_info.pid = pid_tgid >> 32;               \
+    E->u.syscall_info.tid = pid_tgid & 0xFFFFFFFF;               \
+    E->u.syscall_info.ppid = -1;               \
+    E->u.syscall_info.luid = -1;               \
+    E->u.syscall_info.euid = bpf_get_current_uid_gid() >> 32;             \
+    E->u.syscall_info.egid = bpf_get_current_uid_gid() & 0xFFFFFFFF;             \
+    E->u.syscall_info.mono_ns = bpf_ktime_get_ns();       \
+    E->u.syscall_info.syscall_pattern = SP;    \
+    bpf_get_current_comm(E->u.syscall_info.comm, sizeof(E->u.syscall_info.comm));
+
+#define FILL_TELEMETRY_SYSCALL_RET(E, SP) \
+
+
+#define GET_OFFSETS_4_8 \
+    /* if "loaded" doesn't exist in the map, we get NULL back and won't read from offsets                           \
+     * when offsets are loaded into the offsets map, "loaded" should be given any value                             \
+     */                                                                                                             \
+    u64 loaded = 0xec6642829d632573; /* CRC64 of "loaded"  */                                                       \
```

Comment:
can we move the CRC constants to their own header file? that way we can autogenerate that header file in the future

Reply:
good idea

Reply:
Added an `offsets.h` file with these values as constants.

750d697f2242e69233ef10fb19251be28b8c7bae